### DISCUSSION TITLE
Inversion - Abort criteria and ways to achieve good results

### DISCUSSION THREAD

**Discussion by TheNablaOperator** [gimli-org/pyGIMLi#477] [Q&A]
Hey pg-community.

I have a question concerning the abort criteria for the inversion...
The inversion stops the process when reaching one of those two criterias (by default): 1.- Chi² < 1 or 2.- dPhi < 2.

As there are many models which can fit the data, is it possible to say which criteria is better to reach? Is it possible to say, a result Chi²=1.2, dPhi=5 is better than Chi²=6, dPhi=0.5 ?

And the second question would be, how to influence that...?
Currently I change the parameter in the inversion, such as:
- lam
- zWeight

Currently I am trying some subdomain regularizations, as mentioned in transform2022 (Lake example)
Are there any other values to "play with"?

Thank you in advance!

**Comment by halbmy** [comment]
Assuming you have a good idea of your data errors, a chi-square value close to 1 is always best. If not, it can converge either below or above 1, and then mostly the dPhi (relative change of the global objective function) criterion avoids unnecessary iterations (also restricted by maxIter).

In this case you're probably underestimating your data errors.

If chi=1 is easily reached it might be a hint for overestimated errors so that you could fit your data better (associated with chi<1).

We have to document this much more, maybe when redesigning the inversion.

**Reply by TheNablaOperator** [reply]
OK... Understood. 
But which parameter can I then use to control that... Let's say I end up with chi²=5, but want to reach further towards chi² -> 1?

**Comment by halbmy** [comment]
If you end up with chi^2=5 you will probably have to decrease your regularization by either lowering `lam` or `zWeight` or any more sophisticated procedure like decoupling, region-wise treatment etc.

**Comment by TheNablaOperator** [comment]
Ok thanks.
I gave that a try...
I simulated one ERT line on a 3D world. Since there is a known structure underneath, I want to restrict that area (like the Lake example). Therefore I added a rectangle with marker=3 and restricted the limits. But it seems that it's ignored by the inversion-algorithm....?
I have values smaller than 90Ohm.m in region 3... And values higher than 30Ohm.m in region 2...
May I ask, what I missed?

pg-version: 1.2.6

```Python
# "het" is generated with ert.simulate()
 
# inversion w\ underground information
mgr = ert.ERTManager(het)
inv2D = mgr.invert(verbose=True, lam=20, zWeight=0.5, paraMaxCellSize=1)

# Addding Information of the underground
plc = mt.createParaMeshPLC(het, paraMaxCellSize=1)
rect = mt.createRectangle(start=[-5, -2], end=[5, 0], marker=3, boundaryMarker=-2)
plc += rect

# Creating the mesh
mesh2 = mt.createMesh(plc, quality=34)

# inversion
mgr2 = ert.ERTManager()
mgr2.inv.setRegularization(2, limits=[90, 300], trans="log")
mgr2.inv.setRegularization(3, limits=[10, 30], trans="log")
mgr2.invert(data=het, mesh=mesh2, verbose=True, lam=20, zWeight=0.5, paraMaxCellSize=1)
```

![grafik](https://user-images.githubusercontent.com/62592621/210800862-d004f20d-9a33-4e79-ac20-486681d21115.png)


![grafik](https://user-images.githubusercontent.com/62592621/210800697-c33a7dfd-6a98-436c-a57e-6249f0e2f3c8.png)

**Reply by halbmy** [reply]
can you save het and attach it here so that we can run it?

I also encourage you to run it with the newest version 1.3.1.

**Reply by TheNablaOperator** [reply]
Attached you find het.dat as het.txt
[het.txt](https://github.com/gimli-org/gimli/files/10355615/het.txt)

Yes, sure. I am using conda on Win. and had some issues on my nb... :(
